### ISSUE TITLE
Redirects no longer work consistently

### ISSUE THREAD

**Issue by JL102** [FIRSTTeam102/ScoringApp-Serverless#104]
After merging #103, `?` (query) parameters do not consistently work in redirects.

To reproduce:
1. Go to a teamintel page (/reports/teamintel?team_key=102)
2. Navigate to Configure Report Columns
3. Click Save column selections
4. You will receive a 404 page instead of navigating back to teamintel, because the `?` in the url is now encoded

**Comment by caburum** [comment]
I don't think it was from #103 as the generation of redirect URLs wasn't touched.

I think the issue is a combination of [this change](https://github.com/FIRSTTeam102/ScoringApp-Serverless/commit/5378b651edefc1cdd30b597c864e5f80fa42bf41#diff-3d38007a44d0c167dbb1e3bd291a2b74c66c70eba8c35568a1b2d7f7e2de8ffb) and [this](https://github.com/FIRSTTeam102/ScoringApp-Serverless/blob/master/primary/src/routes/user.ts#L626).

In the first one we want to decode the redirect URL so we can go to it and fixRedirectUrl seems to be encoding it again.

In the second, since it's just getting set as form element value and not a query param, it probably shouldn't be encoded at all and submitting the form brings that value to the first one.
![in the form](https://user-images.githubusercontent.com/29152895/178838411-cde738bf-7c83-4448-b60f-84f4a54de1ce.png)

"fixRedirectUrl" is a pretty confusing name as both URL-encoded and decoded could be desired depending on if it's something getting sent to the client or something getting returned from them. You would want to "fix" it on the way out so that the query params from it don't bleed out and you'd want to "fix" it on the way in so you can redirect to it with the params.
